testsuite/features/support/commonlib.rb

```diff
@@ -328,3 +328,100 @@ def add_context(key, value)
   $context[$feature_scope] = {} unless $context.key?($feature_scope)
   $context[$feature_scope].merge!({ key => value })
 end
+
+# This function gets the system name, as displayed in systems list
+# * for the usual clients, it is the full hostname, e.g. suma-41-min-sle15.tf.local
+# * for the PXE booted clients, it is derived from the branch name, the hardware type,
+#   and a fingerprint, e.g. example.Intel-Genuine-None-d6df84cca6f478cdafe824e35bbb6e3b
+def get_system_name(host)
+  case host
+    # The PXE boot minion and the terminals are not directly accessible on the network,
+    # therefore they are not represented by a twopence node
+  when 'pxeboot_minion'
+    output, _code = get_target('server').run('salt-key')
+    system_name =
+      output.split.find do |word|
+        word =~ /example.Intel-Genuine-None-/ || word =~ /example.pxeboot-/ || word =~ /example.Intel/ || word =~ /pxeboot-/
+      end
+    system_name = 'pxeboot.example.org' if system_name.nil?
+  when 'sle12sp5_terminal'
+    output, _code = get_target('server').run('salt-key')
+    system_name =
+      output.split.find do |word|
+        word =~ /example.sle12sp5terminal-/
+      end
+    system_name = 'sle12sp5terminal.example.org' if system_name.nil?
+  when 'sle15sp4_terminal'
+    output, _code = get_target('server').run('salt-key')
+    system_name =
+      output.split.find do |word|
+        word =~ /example.sle15sp4terminal-/
+      end
+    system_name = 'sle15sp4terminal.example.org' if system_name.nil?
```

Comment:
can't these be simplified above, or am I missing something?

Reply:
Maybe, but out of scope of this PR.